interpreter/cling/lib/Interpreter/DefinitionShadower.cpp

```diff
@@ -0,0 +1,189 @@
+//------------------------------------------------------------------------------
+// CLING - the C++ LLVM-based InterpreterG :)
+// author:  Javier López-Gómez <jalopezg@inf.uc3m.es>
+//
+// This file is dual-licensed: you can choose to license it under the University
+// of Illinois Open Source License or the GNU Lesser General Public License. See
+// LICENSE.TXT for details.
+//------------------------------------------------------------------------------
+
+#include "DefinitionShadower.h"
+
+#include "cling/Interpreter/Interpreter.h"
+#include "cling/Interpreter/InterpreterCallbacks.h"
+#include "cling/Utils/AST.h"
+
+#include "clang/AST/Decl.h"
+#include "clang/AST/DeclContextInternals.h"
+#include "clang/AST/Stmt.h"
+#include "clang/Sema/Lookup.h"
+#include "clang/Sema/Sema.h"
+
+using namespace clang;
+
+namespace cling {
+  /// \brief Returns whether the given source location is a Cling input line. If
+  /// it came from the prompt, the file is a virtual file with overriden contents.
+  static bool typedInClingPrompt(FullSourceLoc L) {
+    if (L.isInvalid())
+      return false;
+    const SourceManager &SM = L.getManager();
+    const FileID FID = SM.getFileID(L);
+    return SM.isFileOverridden(SM.getFileEntryForID(FID))
+           && (SM.getFileID(SM.getIncludeLoc(FID)) == SM.getMainFileID());
+  }
+
+  /// \brief Returns whether the given {Function,Tag,Var}Decl/TemplateDecl is a definition.
+  static bool isDefinition(const Decl *D) {
+    if (auto FD = dyn_cast<FunctionDecl>(D)) return FD->isThisDeclarationADefinition();
```

Comment:
Can you make this two lines each?
```suggestion
    if (auto FD = dyn_cast<FunctionDecl>(D))
      return FD->isThisDeclarationADefinition();
```

Reply:
Sure.  I forgot to split those.  :-(